scripts/devops_tasks/test_regression.py

```diff
@@ -0,0 +1,364 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# This script will run regression test for packages which are added as required package by other packages
+# Regression test ensures backword compatibility with released dependent package versions
+
+import argparse
+import glob
+import sys
+import os
+import logging
+from common_tasks import (
+    process_glob_string,
+    parse_setup,
+    run_check_call,
+    parse_require,
+    install_package_from_whl,
+    filter_dev_requirements,
+    OmmitType,
+)
+from git_helper import get_release_tag, checkout_code_repo, clone_repo
+
+AZURE_GLOB_STRING = "azure*"
+
+root_dir = os.path.abspath(os.path.join(os.path.abspath(__file__), "..", "..", ".."))
+test_tools_req_file = os.path.abspath(os.path.join(root_dir, "eng", "test_tools.txt"))
+
+GIT_REPO_NAME = "azure-sdk-for-python"
+GIT_MASTER_BRANCH = "master"
+VENV_NAME = "regressionenv"
+AZURE_SDK_FOR_PYTHON_GIT_URL = "https://github.com/Azure/azure-sdk-for-python.git"
+TEMP_FOLDER_NAME = ".tmp_code_path"
+COSMOS_TEST_ARG = "not cosmosEmulator"
+
+PACKAGE_RELEASE_TAGS_TO_EXCLUDE = ['azure-storage-file-share:*',]
+
+logging.getLogger().setLevel(logging.INFO)
+
+
+class CustomVirtualEnv:
+    def __init__(self, path):
+        self.path = os.path.join(path, VENV_NAME)
+
+    def create(self):
+        logging.info("Creating virtual environment [{}]".format(self.path))
+        run_check_call([sys.executable, "-m", "venv", "ENV_DIR", self.path], root_dir)
+
+    def _process_venv(self, activate_env):
+        if activate_env:
+            # clear any previously installed packages
+            run_check_call(
+                [sys.executable, "-m", "venv", "--clear", "ENV_DIR", self.path],
+                root_dir,
+            )
+
+        scriptName = "activate.bat" if activate_env else "deactivate.bat"
+        venv_script = os.path.join(self.path, "Scripts", scriptName)
+        operation_type = "Activating" if activate_env else "Deactivating"
+
+        if os.path.exists(venv_script):
+            logging.info(
+                "{0} virtual environment {1}".format(operation_type, venv_script)
+            )
+            run_check_call([venv_script,], root_dir)
+        else:
+            logging.error(
+                "Script to process virtualenv is missing. path [{}]".format(venv_script)
+            )
+            sys.exit(1)
+
+    def activate(self):
+        self._process_venv(True)
+
+    def deactivate(self):
+        self._process_venv(False)
+
+
+class RegressionContext:
+    def __init__(self, whl_dir, tmp_path, is_latest):
+        self.whl_directory = whl_dir
+        self.temp_path = tmp_path
+        self.is_latest_depend_test = is_latest
+        self.venv = CustomVirtualEnv(self.temp_path)
+        self.venv.create()
+        self.python_symlink = os.path.abspath(
```

Comment:
Is it actually a symlink? My understanding was that the python in the venv was a real copy of the exe .

After doing a bit of reading: https://docs.python.org/3/library/venv.html#creating-virtual-environments

It looks like unless you pass it `--symlinks` specifically, it's doing a copy of the python instance.